proposals/lip-0047.md

```diff
@@ -763,19 +747,16 @@ def afterTransactionsExecute(b: Block) -> None:
         roundStartHeight = chainProperties.roundEndHeight - previousLengthValidators + 1
         randomSeed = Random.getRandomBytes(roundStartHeight, previousLengthValidators)
 
-        addresses = []
-        for i in range(length(snapshotStore(0).validators)):
-            addresses[i] =  snapshotStore(0).validators[i].address
-
-        nextValidatorAddresses  = shuffleValidatorsList(addresses, randomSeed)
-        Validators.setValidatorParams(snapshotStore(0).threshold, snapshotStore(0).threshold, nextValidatorAddresses)
+        nextValidators  = PoS.shuffleValidatorsList(snapshotStore(0).validators, randomSeed)
```

Comment:
I find this an undesirable dependency: the idea is that chains should use PoA OR PoS, so it does not make much sense IMO to call a PoS endpoint from PoA module. 

Perhaps the best would be if we move the `shuffleValidatorsList` to random module, but I need to think a bit about it.